src/Database/PostgreSQL/PQTypes/Internal/Connection.hs

```diff
@@ -370,8 +396,7 @@ runQueryImpl fname conn sql execSql = do
               , statsValues = statsValues cdStats + (rows * columns)
               , statsParams = statsParams cdStats + paramCount
               }
-      -- Force evaluation of modified stats to squash a space leak.
-      stats' `seq` return (cd {cdStats = stats'}, (either id id affected, res))
+      return (cd {cdStats = stats'}, (either id id affected, res))
```

Comment:
What about using `pure` rather than `return` since it's what you used everywhere else ?

Reply:
Right, although there's tons of return in other places, so whether I replace this one or not is irrelevant. Let's leave it for later when hlint is enabled for this repo, it can be then done systematically for all returnS.